src/components/ens-profile/ActionButtons/MoreButton.tsx

```diff
@@ -27,10 +30,28 @@ export default function MoreButton({
   address?: string;
   ensName?: string;
 }) {
+  const { switchToWalletWithAddress, wallets, selectedWallet } = useWallets();
   const { navigate } = useNavigation();
   const { setClipboard } = useClipboard();
   const { contacts, onRemoveContact } = useContacts();
 
+  const isWatchedOrOwnedWallet = useMemo(() => {
+    return Object.keys(wallets).find(key => {
+      return wallets[key].addresses.some(
+        (wallet: { address: string }) =>
+          wallet.address.toLowerCase() === address?.toLowerCase()
+      );
+    });
+  }, [address, wallets]);
```

Comment:
can we not use the `useWatchWallet` hook here? that returns an `isWatching` value.